shared/githubconnector/get_releases.go

```diff
@@ -0,0 +1,111 @@
+package githubconnector
+
+import (
+	"context"
+	"strings"
+
+	"github.com/google/go-github/v53/github"
+)
+
+func (client *GithubConnector) GetRelease(org, project, version string) (*Release, error) {
+
+	// I don't like this, but we have to add it somewhere. Bury it here so everything else can
+	// work on the basis of versions instead of tags.
+	versionTag := "v" + version
+
+	ghRelease, _, err := client.github.Repositories.GetReleaseByTag(context.Background(), org, project, versionTag)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &Release{
+		Tag:     *ghRelease.TagName,
+		Version: strings.Replace(*ghRelease.TagName, "v", "", 1),
+		Assets:  getAssetsForRelease(ghRelease),
+	}, nil
+
+}
+
+func (client *GithubConnector) GetReleases(org string, project string) ([]*Release, error) {
+
+	happyReleases := make([]*Release, 0)
+
+	// Only up to 1000 results are returned by the API
+	for page := 1; page < 2; page++ {
```

Comment:
do the pages work like this? I thought we had to parse the page out of the resp like

~~~
page = resp.NextPage
~~~

Reply:
I'm only getting one page here. I was getting back all the responses, but they really weren't necessary, and I exhausted my API call limit for my IP the hour. The tool doesn't (yet?) authenticate with a PAT, so the limit is rather low.

Might later add an option to get more pages if people care.

As far as parsing the response, I'm using the Github client rather than a raw HTTP client, so I guess it's probably doing that for me. This is how the client I'm using works.

Reply:
Also, this will probably have to be configurable in the function call, since this is in the shared/ module and could be used for other stuff. Not going to bother with it just yet, though.